### PR TITLE
Inline JS for reports

### PR DESCRIPTION
**Pull request by divs1210**
[Story on Pivotal](https://www.pivotaltracker.com/story/show/163451861)

### CODE REVIEW

File: app/controllers/api/current/analytics/user_analytics_controller.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ class Api::Current::Analytics::UserAnalyticsController < Api::Current::Analytics
 
   def show
     @stats_for_user = new_patients_by_facility_week
+    @application_js = asset_source('application.js')
```
[the remark this answers is not in the document]
Comment (divs1210) [reply app/controllers/api/current/analytics/user_analytics_controller.rb]: How do you suggest we do this?
I was unable to find clear answers on the web.
We are using this [charting library](https://github.com/ankane/chartkick).

File: app/controllers/api/current/analytics/user_analytics_controller.rb [hunk 2]
```diff
@@ -21,4 +22,13 @@ def new_patients_by_facility_week
       .group_by_week('device_created_at', last: WEEKS_TO_REPORT)
       .count
   end
+
+  def asset_source(asset_path)
+    asset = Rails.application.assets.find_asset(asset_path)
```
[the remark this answers is not in the document]
Comment (divs1210) [reply app/controllers/api/current/analytics/user_analytics_controller.rb]: I've pushed a patch to make this clearer.

File: app/controllers/api/current/analytics/user_analytics_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (timcheadle) [reply app/controllers/api/current/analytics/user_analytics_controller.rb]: If we're going to stick with the asset pipeline, let's at least put it in its own JS file and not bundle the entire `application.js`

File: app/controllers/api/current/analytics/user_analytics_controller.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (timcheadle) [reply app/controllers/api/current/analytics/user_analytics_controller.rb]: I see this on staging:

```
deploy@api-staging:~/apps/simple-server/current$ bundle exec rails c production
Loading production environment (Rails 5.1.6.1)
[1] pry(main)> Rails.application.assets.find_asset("application.js")
NoMethodError: undefined method `find_asset' for nil:NilClass
from (pry):1:in `__pry__'
[2] pry(main)> Rails.application.assets
=> nil
[3] pry(main)> Rails.application.config.assets.compile
=> false
```

How did you verify this works?

[the remark this answers is not in the document]
Comment (divs1210) [reply app/controllers/api/current/analytics/user_analytics_controller.rb]: Hmm. My bad, must have done something wrong.
This code was adapted from [this gist](https://gist.github.com/averyvery/6e4576023b395de1aaf5).
I have updated this PR to only load the minified charting scripts from `vendor/assets`.

**Comment by timcheadle** [comment]
I've taken a stab at cleaning this up. Rather than rely on vendor assets, I made a separate `charts.js` asset and written a helper method to clean up the view and controller.

**Pull request merged**